### PR TITLE
[0.17] Bugfix: Correctly calculate balances when min_conf is used, and for getbalance("*")

### PR DESCRIPTION
**Pull request by luke-jr**
0.17 backport of #14602, but with named-arg account="*" compatibility

### CODE REVIEW

**Comment by MeshCollider** [comment]
Concept ACK

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -2110,17 +2111,24 @@ void CWallet::ResendWalletTransactions(int64_t nBestBlockTime, CConnman* connman
...
         for (const auto& entry : mapWallet)
         {
             const CWalletTx* pcoin = &entry.second;
-            if (pcoin->IsTrusted() && pcoin->GetDepthInMainChain() >= min_depth) {
-                nTotal += pcoin->GetAvailableCredit(true, filter);
+
+            if (trusted_only) {
+                if (!pcoin->IsTrusted()) continue;
+            } else {
+                if (!CheckFinalTx(*pcoin->tx)) continue;
```
Comment (zallarak) [inline src/wallet/wallet.cpp]: Why not else if?

File: src/wallet/wallet.cpp [hunk 2]
```diff
@@ -2110,17 +2111,24 @@ void CWallet::ResendWalletTransactions(int64_t nBestBlockTime, CConnman* connman
...
     CAmount nTotal = 0;
     {
         LOCK2(cs_main, cs_wallet);
         for (const auto& entry : mapWallet)
         {
             const CWalletTx* pcoin = &entry.second;
-            if (pcoin->IsTrusted() && pcoin->GetDepthInMainChain() >= min_depth) {
-                nTotal += pcoin->GetAvailableCredit(true, filter);
+
+            if (trusted_only) {
```
Comment (zallarak) [inline src/wallet/wallet.cpp]: is the nested `if` preferred over `trusted_only && !pcoin->IsTrusted()`?

**Comment by gmaxwell** [comment]
Concept ACK.

**Comment by jonasschnelli** [comment]
utACK 32de31c1dd3255702f655b0746cb4fac3cc1938f

**Comment by luke-jr** [comment]
The linter is broken by design.

**Pull request closed without merging**